src/filter/filter.c

```diff
@@ -22,28 +22,25 @@
 
 #include "filter.h"
 
-//Implements a fast Exponential Moving Average IIR filter
-
-//This macros defines an alpha value between 0 and 1
-#define DSP_EMA_I32_ALPHA(x) ( (unsigned short)(x * 65535) )
-
-static int dsp_ema_i32(int in, int average, unsigned short alpha)
-{
-    long long tmp0; //calcs must be done in 64-bit math to avoid overflow
-    tmp0 = (long long)in * (alpha) + (long long)average * (65536 - alpha);
-    return (int)((tmp0 + 32768) / 65536); //scale back to 32-bit (with rounding)
-}
-
+//Implements a fast Exponential Moving Average filter
+#define MIN_ALPHA 0.0001
 void init_filter(Filter *filter, float alpha)
 {
-    filter->alpha = alpha;
-    filter->current_value = 0;
+        filter->max_samples = 1 / (alpha != 0 ? alpha : MIN_ALPHA);
+        filter->current_value = 0;
+        filter->total = 0;
+        filter->count = 0;
 }
 
-int update_filter(Filter *filter, int value)
+int32_t update_filter(Filter *filter, int32_t value)
```

Comment:
Should there be be `const` modifiers on the parameters?

Reply:
yep, added to the 2nd param, but not the first, as we're making a change to the member of the struct.